nibabel/openers.py

```diff
@@ -22,6 +22,7 @@
     from types import TracebackType
 
     from _typeshed import WriteableBuffer
+    from typing_extensions import Self
```

Comment:
Need to add this to dependencies. Should guard and import from typing for Python <3.11, and then we can use `typing_extensions >=VER; python_version < '3.11'`.

Reply:
Add `typing_extensions`.